q2_types/per_sample_sequences/tests/test_format.py

```diff
@@ -26,6 +28,97 @@
 from qiime2.plugin import ValidationError
 
 
+class TestAbsoluteFastqManifestV2Formats(TestPluginBase):
+    package = 'q2_types.per_sample_sequences.tests'
+
+    def setUp(self):
+        super().setUp()
+        self.se_formats = [SingleEndFastqManifestPhred33V2,
+                           SingleEndFastqManifestPhred64V2]
+        self.pe_formats = [PairedEndFastqManifestPhred33V2,
+                           PairedEndFastqManifestPhred64V2]
+
+    def template_manifest(self, filepath, ctx):
+        with open(filepath) as fh:
+            tmpl = string.Template(fh.read())
+        basename = os.path.basename(filepath)
+        file_ = os.path.join(self.temp_dir.name, basename)
+        with open(file_, 'w') as fh:
+            fh.write(tmpl.substitute(**ctx))
+        return file_
+
+    def test_validate_se_positive(self):
+        s1 = self.get_data_path('Human-Kneecap_S1_L001_R1_001.fastq.gz')
+        fp = self.get_data_path('absolute_manifests_v2/single-MANIFEST')
+        manifest = self.template_manifest(fp, {'path': s1})
+
+        for fmt in self.se_formats:
+            fmt(manifest, mode='r').validate()
+
+    def test_validate_pe_positive(self):
+        s1 = self.get_data_path('Human-Kneecap_S1_L001_R1_001.fastq.gz')
```

Comment:
Hmm, what if I want to import the same file multiple times as different samples? Not sure why you would want to (outside of my test suite, that is).

Reply:
It wouldn't allow that any more. My thought was that that is probably an error if it happens.

Reply:
Sorry, my hypothetical question wasn't clear. I am proposing that someone might deliberately do this. For example, including the same replicate in a few different sample groups or pools. Not sure why you would do that, though.

Reply:
I don't _think_ you ever would want to do that, except maybe in the case of an expert user/developer who wants to evaluate what happens to some replicated sequence data in a processing workflow. I'm thinking that by explicitly disallowing it you would reduce errors/support burden and that users who really did want to do this could work around it by copying input files so they have different paths in this file. 

Your call though on whether QIIME 2 should allow it or not. It's totally subjective: the balance of protecting users from making mistakes versus empowering users to do experimental things.

Reply:
I decided to add this in to this PR.